src/main/java/frc/robot/subsystems/IntakeSubsystem.java

```diff
@@ -71,13 +87,54 @@ public void close() throws Exception {
     // Mostly for JUnit tests
     motor.close();
     motor2.close();
-    
+  }
+
+  public double getConeDist() {
+    return timeOfFlightSensor.getRawDistance0().classDistance;
+  }
+
+  public double getCubeDist() {
+    return timeOfFlightSensor.getRawDistance1().classDistance;
+  }
+
+  public void changeFlightState() {
+    double coneDistance = getConeDist();
+    double cubeDistance = getCubeDist();
+
+    boolean coneSensorUp = coneDistance != -1;
+    boolean cubeSensorUp = cubeDistance != -1;
+
+    SmartDashboard.putNumber("Cone Distance", coneDistance);
+    SmartDashboard.putNumber("Cube Distance", cubeDistance);
+    SmartDashboard.putBoolean("Has Cone", coneDistance <= coneActivationThreshold);
+    SmartDashboard.putBoolean("Has Cube", cubeDistance <= cubeActivationThreshold);
+    SmartDashboard.putBoolean("Cone Sensor Working", coneSensorUp);
+    SmartDashboard.putBoolean("Cube Sensor Working", cubeSensorUp);
+
+    if (coneSensorUp && cubeSensorUp) {
```

Comment:
I mean, if the mode is cube, and the cube sensor is up...We would want to proceed as normally.

Reply:
State change only ignored if ToF sensor for that game piece is down